sources/healthdog/src/healthdog_test.rs

```diff
@@ -0,0 +1,162 @@
+                is_healthy: true,
+                exit_code: None,
+            })
+        }
+    }
+}
+
+#[test]
+fn send_healthy_ping() {
+    let server = Server::run();
+    let matcher = all_of![
+        request::method_path("GET", "/metrics"),
+        request::query(url_decoded(contains(("sender", "healthdog")))),
+        request::query(url_decoded(contains(("event", "health-ping")))),
+        request::query(url_decoded(contains(("version", "0.4.0")))),
+        request::query(url_decoded(contains(("variant", "aws-k8s-1.16")))),
+        request::query(url_decoded(contains(("arch", "x86_64")))),
+        request::query(url_decoded(contains(("region", "us-east-1")))),
+        request::query(url_decoded(contains(("seed", "2041")))),
+        request::query(url_decoded(contains(("failed_services", "")))),
+        request::query(url_decoded(contains(("is_healthy", "true")))),
+    ];
+    server.expect(Expectation::matching(matcher).respond_with(status_code(200)));
+    let port = server.addr().port();
+    let healthdog = Healthdog::from_parts(
+        Some(Config {
+            metrics_url: format!("http://localhost:{}/metrics", port),
+            send_metrics: true,
+            services: vec![
+                String::from("service_a"),
+                String::from("service_b"),
+                String::from("service_c"),
+            ],
+            aws_region: String::from("us-east-1"),
+            seed: 2041,
+            version_lock: String::from("latest"),
+            ignore_waves: false,
+        }),
+        Some(os_release()),
+        Some(Box::new(MockCheck {})),
+    )
+    .unwrap();
+    healthdog.send_health_ping().unwrap();
+}
+
+#[test]
+fn send_unhealthy_ping() {
+    let server = Server::run();
+    let matcher = all_of![
+        request::method_path("GET", "/metrics"),
+        request::query(url_decoded(contains(("sender", "healthdog")))),
+        request::query(url_decoded(contains(("event", "health-ping")))),
+        request::query(url_decoded(contains(("version", "0.4.0")))),
+        request::query(url_decoded(contains(("variant", "aws-k8s-1.16")))),
+        request::query(url_decoded(contains(("arch", "x86_64")))),
+        request::query(url_decoded(contains(("region", "us-east-1")))),
+        request::query(url_decoded(contains(("seed", "2041")))),
+        request::query(url_decoded(contains((
+            "failed_services",
+            "service_afail2:2,service_cfail1:1"
+        )))),
+        request::query(url_decoded(contains(("is_healthy", "false")))),
+    ];
+    server.expect(Expectation::matching(matcher).respond_with(status_code(200)));
+    let port = server.addr().port();
+    let healthdog = Healthdog::from_parts(
+        Some(Config {
+            metrics_url: format!("http://localhost:{}/metrics", port),
+            send_metrics: true,
+            // note that these are out-of-order sort order to ensure that failed services are sorted
+            // in the url
```

Comment:
I think so. Although we can't imagine it every happening, if one ever needed to pour over raw records at 3 AM, it would be a lot easier to spot a pattern. The cost of the sort is infinitesimal so I would vote for leaving in favor of deterministic urls.